### ISSUE TITLE
Foxx Controller: value of req.server.address

### ISSUE THREAD

**Issue by stackmagic** [arangodb/arangodb#1190]
(from #1170) It's debatable whether my initial thoughts on this are correct.

The field `req.server.address` contains `0.0.0.0` in my case. And that's probably from the endpoint definition in `arangod.conf` - so this may be as designed after all. Realizing that an arangod instance may expose multiple endpoints it does make sense to put this information into the request instead of the application context.

The way I now get the app's proto/server/port/basepath to build url's to send back to the client is this (it's even safe for use behind proxies):

```javascript
exports.getBasePath = function(req) {
		var proto = req.headers["x-forwarded-proto"] || req.protocol;
		var host  = req.headers["x-forwarded-host"]  || req.headers.host;
		return proto + "://" + host + applicationContext.baseUrl;
	}
```

**Comment by pluma** [comment]
We could probably implement something like `req.protocol` and `req.host` (or something to that effect) in the request object and have it behave the way you described (i.e. use the proxy headers or the defaults).

I'm not sure whether allowing proxy-headers to override this is a security concern. Some web frameworks have a flag to toggle this behaviour but others seem to just use them by default.

**Comment by pluma** [comment]
This will be addressed in the 2.6 release. Currently there's an `absoluteUrl` method on the request object in devel that performs the logic you described. Any further improvements will be addressed via #1191.
